src/include/migraphx/op/gather.hpp

```diff
@@ -125,13 +125,12 @@ struct gather
                     auto out_lens  = data.get_shape().lens();
                     out_lens[axis] = indices.get_shape().elements();
                     migraphx::shape out_comp_shape{data.get_shape().type(), out_lens};
-                    shape_for_each(out_comp_shape, [&](const auto& out_idx) {
-                        auto data_idx  = out_idx;
-                        auto in_index  = indices[data_idx[axis]];
-                        in_index       = (in_index < 0) ? in_index + axis_dim_size : in_index;
-                        data_idx[axis] = in_index;
-                        output[out_comp_shape.index(out_idx.begin(), out_idx.end())] =
-                            data(data_idx.begin(), data_idx.end());
+                    shape_for_each(out_comp_shape, [&](size_t out_idx, const auto& out_idx_v) {
+                        auto data_idx   = out_idx_v;
+                        auto in_index   = indices[data_idx[axis]];
+                        in_index        = (in_index < 0) ? in_index + axis_dim_size : in_index;
+                        data_idx[axis]  = in_index;
+                        output[out_idx] = data(data_idx.begin(), data_idx.end());
```

Comment:
It will be faster to do `output(out_idx_v)`, as that will avoid divisions in the index calculations.

Reply:
Ah wait we dont have an overload for vector directly. So you need to do `output(out_idx_v.begin(), out_idx_v.end())`.

Reply:
`out_idx` is the (equivalent of) calculation result of `out_comp_shape.index(out_idx.begin(), out_idx.end())` -- that was replaced. I am not clear why do we need to anything else..

Reply:
Using `out_idx_v` will only do multiplications and additions to get the offset:

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/shape.cpp#L349

Using `out_idx` is a more complicated calculation that requires divisions and mods to calculate the offset, so is slower:

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/shape.cpp#L370

Reply:
(Thanks for the explanation, and I now understand your concern..)  

However, the default of `this->standard()` case  : https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/shape.cpp#L359
is just a pass-through (and would be of course the fastest with `out_idx`)

I can revert this, if you like.. ; Thanks.

Reply:
Faster with standard and slower for non-standard. I guess the standard shape is more common for ref operators so it might be better to use `out_idx`.